editor/code_editor.cpp

```diff
@@ -418,6 +418,24 @@ void FindReplaceBar::_update_matches_label() {
 	}
 }
 
+void FindReplaceBar::_temp_unfold_line(int &p_line) {
+	if (temp_unfolded_line != -1) {
+		text_editor->fold_line(temp_unfolded_line);
+	}
+
+	if (!text_editor->is_line_folded(p_line)) {
+		temp_unfolded_line = p_line;
+		while (temp_unfolded_line > 0) {
+			if (text_editor->is_line_folded(temp_unfolded_line)) {
+				break;
+			}
+			temp_unfolded_line--;
+		}
+
+		text_editor->unfold_line(p_line);
```

Comment:
This line isn't folded by the above check, did you mean to use `temp_unfolded_line`?

Also this code doesn't modify `p_line`, is there any reason it is a reference?

Reply:
You got the idea wrong, so I added a comment to clarify my code.

Consider that I have a folded function, which has a loop, a condition, and then some code in the condition, and `p_line`, the line that should be temporarily unfolded, is in the condition code. I will want to walk back and mark that the function is the first line of code that needed to be unfolded so that `p_line` is shown. Therefore, I'll unfold `p_line`, while keeping `temp_unfolded_line` to mark that function is temporarily unfolded.

Reply:
But the line isn't folded, is it?

The question of the reference still remains